presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -104,6 +110,8 @@ public TypeRegistry(Set<Type> types)
         addType(JSON_PATH);
         addType(COLOR);
         addType(JSON);
+        addParametricType(VarcharParametricType.VARCHAR);
+        addParametricType(DecimalParametricType.DECIMAL);
         addParametricType(ROW);
```

Comment:
this is not exactly `subType`. I agree that `typesMatch` is too generic. But i can't find any better name for now.